File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    int count(int n) {
        int c=0;
        while(n) {
            if(n&1) c++;
            n = n>>1;
        }
        return c;
    }
    vector<int> countBits(int n) {
        vector<int>v(n+1);
        for(int i=0; i<v.size(); i++) {
            v[i] = count(i);
        }
        return v;
    }
};

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        a = []
        for i in range(n+1):
            x = i
            c = 0
            while x:
                if x&1:
                    c += 1
                x >>= 1    
            a.append(c)
        return a
                    
        